src/modules/ekf2/EKF/control.cpp

```diff
@@ -362,7 +337,14 @@ void Ekf::controlExternalVisionFusion()
 
 		// determine if we should use the yaw observation
 		if (_control_status.flags.ev_yaw) {
-			fuseHeading();
+			if (shouldUse321RotationSequence(_R_to_earth)) {
```

Comment:
This was moved so that ev_yaw heading fusion couldn't be impacted by the magnetometer `_is_yaw_fusion_inhibited` logic embedded in fuseHeading().